src/gameobjects/Player.cpp

```diff
@@ -295,7 +295,7 @@ void Player::standOrBlockMovement(GameObject *go, int heigth) {
         if (yPosition < floor) {
             yPosition = yPosition + GRAVITY > floor ? floor : yPosition + GRAVITY;
         }
-        if (yPosition > yBlock) {
+        if (yPosition >= (yBlock - go->getFloorPosition() - go->getSize())) {
             jumping = false;
             if(go->getType() == GOT_PLATFORM_SURPRISE) {
```

Comment:
```suggestion
            if(go->getType() == GOT_PLATFORM_SURPRISE && abs(xPosition - go->getXPosition()) < go->getWidth() - 5 ) {
```
El -5 viene a cumplir un rol de asegurarme de que cuando choque en los costados no cuente como golpe